include/deal.II/base/auto_derivative_function.h

```diff
@@ -157,17 +158,19 @@ class AutoDerivativeFunction : public Function<dim>
    * Compute numerical difference quotients using the preset
    * #DifferenceFormula.
    */
-  virtual Tensor<1,dim> gradient (const Point<dim>   &p,
-                                  const unsigned int  component = 0) const override;
+  virtual Tensor<1, dim>
+  gradient(const Point<dim> & p,
+           const unsigned int component = 0) const override;
 
   /**
    * Return the gradient of all components of the function at the given point.
    *
    * Compute numerical difference quotients using the preset
    * #DifferenceFormula.
    */
-  virtual void vector_gradient (const Point<dim>            &p,
-                                std::vector<Tensor<1,dim> > &gradients) const override;
+  virtual void
+  vector_gradient(const Point<dim> &           p,
+                  std::vector<Tensor<1, dim>> &gradients) const override;
```

Comment:
I thought we instructed `clang-format` to put `&`s and `*`s all the way to the right. Is there a way to fix this?

Reply:
No, this is why I voted for `Left`. There is a PR that might make its way into one of the next ` llvm` version to fix it.

Reply:
https://reviews.llvm.org/D27651

Reply:
I now understand why you voted for that option. Lets keep what we have and hope that an option to fix this appears in the future.